fastly/helpers.go

```diff
@@ -31,3 +33,22 @@ func NullString(v string) *string {
 	}
 	return &v
 }
+
+// ToSafeURL produces a safe (no path traversal, no unsafe characters) URL
+// from the path components passed in.
+func ToSafeURL(unsafeComponents ...string) string {
+	safeComponents := make([]string, len(unsafeComponents))
+
+	for i := range unsafeComponents {
+		if component := unsafeComponents[i]; component != ".." {
+			safeComponents[i] = url.PathEscape(component)
```

Comment:
The extra local variable `component` isn’t needed in Go, since a (modified) copy of the input is returned by `PathEscape`.  Perhaps a Rust idiom or habit showing up here?

Reply:
Old habits die hard :-) This was the "don't repeat array indexing" habit.